test/index.js

```diff
@@ -1091,6 +1065,114 @@ describe('PubSub', function() {
     });
   });
 
+  describe('getClient_', function() {
+    var CONFIG = {
+      client: 'SubscriberClient',
+      reqOpts: {},
+      gaxOpts: {},
+    };
+
+    beforeEach(function() {
+      pubsub.config = CONFIG;
+    });
+
+    it('should cache the client', function(done) {
+      pubsub.getClient_(CONFIG, function(err, client1) {
+        assert.strictEqual(null, err);
+        pubsub.getClient_(CONFIG, function(err, client2) {
+          assert.strictEqual(null, err);
+          assert.strictEqual(client1, client2);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('request', function() {
+    var CONFIG = {
+      client: 'SubscriberClient',
+      method: 'fakeMethod',
+      reqOpts: {a: 'a'},
+      gaxOpts: {},
+    };
+
+    beforeEach(function() {
+      delete pubsub.projectId;
+
+      pubsub.auth = {
+        getProjectId: function(callback) {
+          callback(null, PROJECT_ID);
+        },
+      };
+
+      fakeUtil.replaceProjectIdToken = function(reqOpts) {
+        return reqOpts;
+      };
+
+      pubsub.config = CONFIG;
+    });
+
+    it('should get the project id', function(done) {
+      pubsub.auth.getProjectId = function(callback) {
+        assert.strictEqual(typeof callback, 'function');
+        done();
+      };
+
+      pubsub.request(CONFIG, assert.ifError);
+    });
+
+    it('should return auth errors to the callback', function(done) {
+      var error = new Error('err');
+
+      pubsub.auth.getProjectId = function(callback) {
+        callback(error);
+      };
+
+      pubsub.request(CONFIG, function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should replace the project id token on reqOpts', function(done) {
```

Comment:
This is correctly placed. (The project ID token replacement happens in `request`)